src/nmodl/nocpout.cpp

```diff
@@ -1988,6 +2098,27 @@ void declare_p() {
         s = ifnew_install("_tsav");
         var_count(s);
     }
+    linsertstr(defs_list,
+               "namespace {\n"
+               "template <typename T>\n"
+               "using _nrn_mechanism_std_vector = std::vector<T>;\n"
+               "using _nrn_model_sorted_token = neuron::model_sorted_token;\n"
+               "using _nrn_mechanism_cache_range = "
+               "neuron::cache::MechanismRange<number_of_floating_point_variables, "
+               "number_of_datum_variables>;\n"
+               "using _nrn_mechanism_cache_instance = "
+               "neuron::cache::MechanismInstance<number_of_floating_point_variables, "
+               "number_of_datum_variables>;\n"
+               "template <typename T>\n"
+               "using _nrn_mechanism_field = neuron::mechanism::field<T>;\n"
+               "template <typename... Args>\n"
+               "void _nrn_mechanism_register_data_fields(Args&&... args) {\n"
+               "  neuron::mechanism::register_data_fields(std::forward<Args>(args)...);\n"
```

Comment:
just a `nitpick`: some of these could be put in common, included header?

Reply:
Some of them (not the ones including `number_of_floating_point_variables` and `number_of_datum_variables`) could be, yeah. I think the rationale was that all of these `_`-prefixed things are `nocmodl`-specific workarounds to avoid clashes in `nocmodl`'s generated code with variable names in MOD files, so they should vanish when `src/nmodl` does. In principle, NMODL would not need them.